.github/workflows/e2e-tests.yml

```diff
@@ -68,6 +68,7 @@ jobs:
         with:
           name: collector-binary
           path: bin/
+      - run: chmod +x bin/*
```

Comment:
Noticed [flaky permissions on the binary](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/7243460577/job/19730526362#step:11:29)